### ISSUE TITLE
Exception in plugin (4.0.3)

### ISSUE THREAD

**Issue by tanmatra** [JetBrains/idea-gitignore#664]
```
Access is allowed from event dispatch thread with IW lock only.

com.intellij.diagnostic.PluginException: Cannot create class mobi.hsz.idea.gitignore.settings.IgnoreSettingsConfigurable [Plugin: mobi.hsz.idea.gitignore]
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:725)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.instantiateClass(ExtensionComponentAdapter.java:51)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.java:133)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:43)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:69)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:501)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:435)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:261)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensions(ExtensionPointName.java:38)
	at com.intellij.openapi.vcs.configurable.VcsManagerConfigurable.buildConfigurables(VcsManagerConfigurable.java:124)
	at com.intellij.openapi.options.SearchableConfigurable$Parent$Abstract.getConfigurables(SearchableConfigurable.java:96)
	at com.intellij.openapi.options.ex.ConfigurableWrapper$CompositeWrapper.getConfigurables(ConfigurableWrapper.java:288)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.getConfigurables(ShowSettingsUtilImpl.java:71)
	at com.intellij.ide.util.gotoByName.GotoActionModel.lambda$new$0(GotoActionModel.java:91)
	at com.intellij.openapi.util.VolatileNotNullLazyValue$1.compute(VolatileNotNullLazyValue.java:56)
	at com.intellij.openapi.util.VolatileNotNullLazyValue.getValue(VolatileNotNullLazyValue.java:36)
	at com.intellij.ide.util.gotoByName.GotoActionModel.getConfigurablesNames(GotoActionModel.java:378)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.processOptions(GotoActionItemProvider.java:176)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.filterElements(GotoActionItemProvider.java:126)
	at com.intellij.ide.actions.searcheverywhere.ActionSearchEverywhereContributor.fetchWeightedElements(ActionSearchEverywhereContributor.java:96)
	at com.intellij.ide.actions.searcheverywhere.MixedResultsSearcher$ContributorSearchTask.run(MixedResultsSearcher.java:179)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.ConcurrencyUtil.lambda$underThreadNameRunnable$3(ConcurrencyUtil.java:196)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:270)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:684)
	... 33 more
Caused by: com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@6b1a6ee5
Current thread: Thread[SE-SearchTask-ActionSearchEverywhereContributor,4,Idea Thread Group] 1354724176
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,Idea Thread Group] 456043531
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1088)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1063)
	at com.intellij.openapi.editor.impl.EditorImpl.assertIsDispatchThread(EditorImpl.java:3301)
	at com.intellij.openapi.editor.impl.EditorImpl.<init>(EditorImpl.java:345)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:197)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:172)
	at mobi.hsz.idea.gitignore.util.Utils.createPreviewEditor(Utils.kt:127)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel$EditorPanel.<init>(IgnoreSettingsPanel.kt:399)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.createUIComponents(IgnoreSettingsPanel.kt:105)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.$$$setupUI$$$(IgnoreSettingsPanel.kt)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.<init>(IgnoreSettingsPanel.kt:62)
	at mobi.hsz.idea.gitignore.settings.IgnoreSettingsConfigurable.<init>(IgnoreSettingsConfigurable.kt:20)
	... 38 more
```
[threadDump.txt](https://github.com/JetBrains/idea-gitignore/files/5987084/threadDump.txt)

**Comment by tanmatra** [comment]
Can it be fixed?

**Comment by Spurlos** [comment]
Stumbled upon same issue, but with .ignore version `4.1.0`. Log from console:

```
2021-04-14 19:55:33,854 [2574689]  ERROR - nsions.impl.ExtensionPointImpl - Cannot create class mobi.hsz.idea.gitignore.settings.IgnoreSettingsConfigurable (classloader=PluginClassLoader(plugin=PluginDescriptor(name=.ignore, id=mobi.hsz.idea.gitignore, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/PhpStorm2021.1/.ignore, version=4.1.0, package=null), packagePrefix=null, instanceId=62, state=active)) 
com.intellij.diagnostic.PluginException: Cannot create class mobi.hsz.idea.gitignore.settings.IgnoreSettingsConfigurable (classloader=PluginClassLoader(plugin=PluginDescriptor(name=.ignore, id=mobi.hsz.idea.gitignore, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/PhpStorm2021.1/.ignore, version=4.1.0, package=null), packagePrefix=null, instanceId=62, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:763)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.instantiateClass(XmlExtensionAdapter.java:94)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.java:112)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:73)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:502)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:436)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:268)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensions(ExtensionPointName.java:35)
	at com.intellij.openapi.vcs.configurable.VcsManagerConfigurable.buildConfigurables(VcsManagerConfigurable.java:124)
	at com.intellij.openapi.options.SearchableConfigurable$Parent$Abstract.getConfigurables(SearchableConfigurable.java:94)
	at com.intellij.openapi.options.ex.ConfigurableWrapper$CompositeWrapper.getConfigurables(ConfigurableWrapper.java:285)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.getConfigurables(ShowSettingsUtilImpl.java:71)
	at com.intellij.ide.util.gotoByName.GotoActionModel.lambda$new$0(GotoActionModel.java:88)
	at com.intellij.openapi.util.NotNullLazyValue$3.compute(NotNullLazyValue.java:80)
	at com.intellij.openapi.util.VolatileNotNullLazyValue.getValue(VolatileNotNullLazyValue.java:19)
	at com.intellij.ide.util.gotoByName.GotoActionModel.getConfigurablesNames(GotoActionModel.java:379)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.processOptions(GotoActionItemProvider.java:166)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.filterElements(GotoActionItemProvider.java:116)
	at com.intellij.ide.actions.searcheverywhere.ActionSearchEverywhereContributor.fetchWeightedElements(ActionSearchEverywhereContributor.java:97)
	at com.intellij.ide.actions.searcheverywhere.MixedResultsSearcher$ContributorSearchTask.run(MixedResultsSearcher.java:177)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.ConcurrencyUtil.lambda$underThreadNameRunnable$3(ConcurrencyUtil.java:201)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:265)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:722)
	... 32 more
Caused by: com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@74f331ca
Current thread: Thread[SE-SearchTask-ActionSearchEverywhereContributor,4,Idea Thread Group] 218988802
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,Idea Thread Group] 898907883
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1047)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1022)
	at com.intellij.openapi.editor.impl.EditorImpl.assertIsDispatchThread(EditorImpl.java:3291)
	at com.intellij.openapi.editor.impl.EditorImpl.<init>(EditorImpl.java:342)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:198)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:173)
	at mobi.hsz.idea.gitignore.util.Utils.createPreviewEditor(Utils.kt:127)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel$EditorPanel.<init>(IgnoreSettingsPanel.kt:400)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.createUIComponents(IgnoreSettingsPanel.kt:106)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.$$$setupUI$$$(IgnoreSettingsPanel.kt)
	at mobi.hsz.idea.gitignore.ui.IgnoreSettingsPanel.<init>(IgnoreSettingsPanel.kt:62)
	at mobi.hsz.idea.gitignore.settings.IgnoreSettingsConfigurable.<init>(IgnoreSettingsConfigurable.kt:21)
	... 37 more
2021-04-14 19:55:33,855 [2574690]  ERROR - nsions.impl.ExtensionPointImpl - PhpStorm 2021.1  Build #PS-211.6693.120 
2021-04-14 19:55:33,859 [2574694]  ERROR - nsions.impl.ExtensionPointImpl - JDK: 11.0.10; VM: Dynamic Code Evolution 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2021-04-14 19:55:33,860 [2574695]  ERROR - nsions.impl.ExtensionPointImpl - OS: Linux 
2021-04-14 19:55:33,860 [2574695]  ERROR - nsions.impl.ExtensionPointImpl - Plugin to blame: .ignore version: 4.1.0 
2021-04-14 19:55:33,860 [2574695]  ERROR - nsions.impl.ExtensionPointImpl - Last Action: EditorBackSpace 
```

**Comment by ForNeVeR** [comment]
Still reproduces in 4.1.1, according to user reports.

**Comment by niktrop** [comment]
@hsz It is one of the most frequent reported exceptions, see 
https://web.ea.pages.jetbrains.team/#/issue/251625

**Comment by hsz** [comment]
Fixed with 54b8138e5c8d64244c02fb092511c11dff0625a1
